query-connector/src/app/backend/usergroup-management.ts

```diff
@@ -163,158 +339,283 @@ export async function saveUserGroupMembership(
 }
 
 /**
- * Retrieves the query group membership for a specific user group.
- * @param groupId - The unique identifier of the user group.
- * @returns A list of queries associated with the specified group.
+ * Retrieves all registered user groups in Query Connector along with member and query counts.
+ * @returns A list of user groups registered in the query connector.
  */
-export async function getQueriesWithGroupStatus(
-  groupId: string,
-): Promise<Query[]> {
+export async function getAllUserGroups(): Promise<QCResponse<UserGroup>> {
```

Comment:
nit: Can this be named `getUserGroups` to align better with REST naming conventions?